packages/dev/core/src/Debug/physicsViewer.ts

```diff
@@ -149,22 +149,32 @@ export class PhysicsViewer {
      */
     protected _updateDebugMeshesV2(): void {
         const plugin = this._physicsEnginePlugin as IPhysicsEnginePluginV2;
-        for (let i = 0; i < this._numBodies; i++) {
+        for (let i = 0; i < this._numBodies; ) {
             const body = this._bodies[i];
+            if (body && body.isDisposed) {
+                this.hideBody(body);
+                continue;

```

Comment:
i is not incremented, so we are stucked in the loop if we take this codepath.

Maybe do something like this for the loop:
```typescript
for (let i = 0; i < this._numBodies; i++) {
    const body = this._bodies[i];
    if (!body) {
        continue;
    }
    if (body.isDisposed) {
        this.hideBody(body);
    } else {
        const transform = this._bodyMeshes[i];
        if (transform) {
            plugin.syncTransform(body, transform);
        }
    }
}
```

Reply:
i is incremented 4 lines below. and hideBody/hideInertia decrements `this._numBodies--;` `this._numInertiaBodies--;`.

Reply:
I've reworked the code a little to make it more clear.

Reply:
Looking at the hideBody function, it seems like it doesn't break after `this._numBodies--`. Is it possible it will be called twice?

Reply:
like here ? https://github.com/BabylonJS/Babylon.js/blob/ba369f7b2d8b8b13956383e508ef9749a4f12125/packages/dev/core/src/Debug/physicsViewer.ts#L503